src/d3a/models/area/__init__.py

```diff
@@ -109,6 +143,29 @@ def __init__(self, name: str = None, children: List["Area"] = None,
             if external_connection_available is True else None
         self.should_update_child_strategies = False
 
+    @property
+    def children(self):
+        return self.__children
+
+    @children.setter
+    def children(self, new_children: AreaChildrenList):
+        if type(new_children) == list:
+            self.__children = AreaChildrenList(self, new_children)
+        elif type(new_children) == AreaChildrenList:
+            new_children.parent_area = self
+            self.__children = new_children
```

Comment:
you probably mean:
```suggestion
            self.children.append(new_children)
```
right?

Reply:
@hannesdiedrich I made the children attribute protected `__children` , so I'm setting it in here .

Reply:
No wait, I a confused. Why do you need the whole block L146-156?

Reply:
This is the case when we have another list of children and we set it to a different parent , I had to set their parent area in here

Reply:
Let's suppose a parent A has Children A , if we had parent B and we set parent B.children = parentA.children , here I'm taking into consideration that we should change the parent of the list

Reply:
IMO this is not a case we will ever have. We only change children of parents, never parents of children. 
(you dont Houses from Community to Community). But maybe the others can comment on that.

Reply:
Thinking from the electrical network perspective, it indeed is a possibility to `change parent of children`. Grid Operator could do so to balance two neighbouring radial feeder or to balance load in between transformers or simply to switch tail-end houses to other feeders because they were experience poor voltage regulation.

TLDR: Its indeed a possibility but I wouldn't be in favour of incorporating the targeted code changes so silently and would make a ticket as well as discuss with research team if any immediate need to do so.

Reply:
@faizan2590 right , I'm open to opening a new ticket for this matter
@spyrostz can we have your opinion on this

Reply:
@faizan2590 I created a ticket and assigned it to the research team https://gridsingularity.atlassian.net/browse/D3ASIM-3229

Reply:
@faizan2590 https://gridsingularity.atlassian.net/browse/D3ASIM-3229?focusedCommentId=21143

Reply:
@Hassan754 AFAIK, protected variables are only prepended with one underscore, not with two, right?

Reply:
> The double underscore “__” (called “dunder“) is used to make an instance attribute or method private (cannot be accessed from outside the class) — when used as leading dunder. When used as enclosing dunder (e.g. “__init__”) it indicates that it is a special method in Python (called “magic method”).

Reply:
Interesting approach in general, but there are 2 major problems with this:
1. there is no need to keep track of the parent in the children list. Each Area instance has a parent member, which already keeps a reference to the parent of this area (it is None for the root of the tree)
2. I highly doubt that we are ever going to move areas from one parent to another, because this might introduce cycles in the tree, and the whole code operates under the assumption that the graph is acyclic (tree is by definition acyclic). If we introduce a graph with cycles, I am certain that multiple parts of d3a will not work (first and foremost, offer/bid forwarding will endlessly forward the same offers and bids in the graph cycle), therefore if we ever need to implement this, we should do this much more carefully. 
For now I personally do not see a strong reason to implement it, even from the electricity network perspective, it will be much easier and far less error prone to destroy and recreate the area objects rather than swap their references. Therefore I would recommend to delete this property altogether.

Reply:
@spyrostz Removed the property